src/lib/ares_dns_parse.c

```diff
@@ -1067,6 +1065,16 @@ static ares_status_t ares_dns_parse_rr(ares__buf_t *buf, unsigned int flags,
     type = ARES_REC_TYPE_RAW_RR;
   }
 
+  if (sect == ARES_SECTION_ANSWER && (flags & ARES_DNS_PARSE_ANSWER_RR_RAW)) {
+    type = ARES_REC_TYPE_RAW_RR;
+  }
+  if (sect == ARES_SECTION_AUTHORITY && (flags & ARES_DNS_PARSE_AUTHORITY_RR_RAW)) {
+    type = ARES_REC_TYPE_RAW_RR;
+  }
+  if (sect == ARES_SECTION_ADDITIONAL && (flags & ARES_DNS_PARSE_ADDITIONAL_RR_RAW)) {
+    type = ARES_REC_TYPE_RAW_RR;
+  }
+
```

Comment:
I'd prefer to see this split into 6 total flags, or reduced to 2.  I know for your use case this is sufficient but I don't want to be locked in.  I'd like to see a split based on `ares_dns_rec_type_allow_name_compression(type)`, so either:

```
ARES_DNS_PARSE_AN_BASE_RAW /* Answers from RFC 1035 that allow name compression */
ARES_DNS_PARSE_NS_BASE_RAW /* Authority from RFC 1035 that allow name compression */
ARES_DNS_PARSE_AR_BASE_RAW /* Additional from RFC 1035 that allow name compression */
ARES_DNS_PARSE_AN_EXT_RAW /* Answers from later RFCs (no name compression) */
ARES_DNS_PARSE_NS_EXT_RAW /* Authority from later RFCs (no name compression) */
ARES_DNS_PARSE_AR_BASE_RAW /* Additional from later RFCs (no name compression) */
```
or 
```
ARES_DNS_PARSE_BASE_RAW /* Records from RFC 1035 that allow name compression */
ARES_DNS_PARSE_EXT_RAW /* Records from later RFCs (no name compression) */
```